File: JeongSH/BOJ/CodePlus.Basic1/Math/boj2609.java
package CodePlus.Basic1.Math;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class boj2609 {
    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] strArr = br.readLine().split(" ");
        int max = Math.max(Integer.parseInt(strArr[0]),Integer.parseInt(strArr[1]));
        int min = Math.min(Integer.parseInt(strArr[0]),Integer.parseInt(strArr[1]));
        int result1 = 0;
        for(int i=min; i>0; i--){
            if((max%i==0)&&(min%i==0)){result1=i; break;}
        }
        int result2 = 0;
        for(int i=1; i<=min ; i++){
            if((max*i)%min==0){result2=i*max; break;}
        }
        System.out.println(result1);
        System.out.println(result2);
    }
}

File: Hyewon711/백준/22-04-07/BOJ-2609.py
# BOJ - 2609 최대공약수와 최소공배수
# 나누었을 때 나머지 0인 값중 가장 큰 값을 최대 공약수 (max) 
# 공배수 중 가장 작은 공배수 (min)
# 파이썬에선 math 모듈에 최대공약수 함수 gcd, 최소공배수 함수 lcm이 존재한다.

a, b = map(int, input().split())

tmp = []
for i in range(1, min(a,b) + 1): # 1부터 두 수중 더 작은 수까지 반복
    if a % i == 0 and b % i == 0: # 나머지가 0인 경우 
        tmp.append(i) # tmp에 추가한다.

print(max(tmp)) # 최대공약수는 tmp에서 가장 큰 값이 된다.
print((a * b) // max(tmp)) # 최소 공배수는 두 수의 곱을 최대공약수로 나눈 몫이 된다. 